### PR TITLE
Remove periodic timer from Scaffold snackbar logic

### PR DESCRIPTION
**Pull request by jonahwilliams**
Also only setState in drawer opened callback if the other drawer type is present

### CODE REVIEW

**Comment by jonahwilliams** [comment]
cc @jslavitz

File: packages/flutter/lib/src/material/scaffold.dart [hunk 1]
```diff
@@ -1023,15 +1023,17 @@ class ScaffoldState extends State<Scaffold> with TickerProviderStateMixin {
   bool _endDrawerOpened = false;
 
   void _drawerOpenedCallback(bool isOpened) {
-    setState(() {
-      _drawerOpened = isOpened;
-    });
+    _drawerOpened = isOpened;
+    if (hasEndDrawer) {
```
Comment (amirh) [inline packages/flutter/lib/src/material/scaffold.dart]: Should we test this?

**Review by amirh — COMMENTED** [review]
Looks good just one more test.

nit: For future PRs I think it's better to have 2 separate PRs for such unrelated changes

File: packages/flutter/lib/src/material/scaffold.dart (hunk 1 above)
Comment (jonahwilliams) [reply packages/flutter/lib/src/material/scaffold.dart]: Probably ... I will move to another PR (so I can confirm that it fixes the benchmarks too)

**Review by amirh — APPROVED** [review]

**Comment by Hixie** [comment]
Thanks.

**Pull request merged**